crates/chain/src/canonical_iter.rs

```diff
@@ -204,52 +235,64 @@ impl<A: Anchor, C: ChainOracle> Iterator for CanonicalIter<'_, A, C> {
     type Item = Result<(Txid, Arc<Transaction>, CanonicalReason<A>), C::Error>;
 
     fn next(&mut self) -> Option<Self::Item> {
```

Comment:
Ideally, we would want most of this called in `CanonicalIter::new`. ~However, I see that it requires us to return an error which would be a breaking change. We also can't store the error in `CanonicalIter` as the error type is a type parameter on `ChainOracle`.~

**Edit:** Let's break the API. `CanonicalIter` no longer makes sense. We should just have a single function that returns `TopologicalIter`/`impl Iterator<Item = ...>`.